### PR TITLE
feat: Add base endpoints for Stripe integration [DEV-3671]

### PR DESCRIPTION
**Pull request by lampkin-diet**

### CODE REVIEW

**Comment by ankurdotb** [comment]
Task linked: [DEV-3671 Backend: Integrate Stripe payments into CaaS portal to provide access to different tiers](https://app.clickup.com/t/6600954/DEV-3671)

**Comment by Tweeddalex** [comment]
Hey @lampkin-diet could you please take a look at why the build is failing here :)

File: src/middleware/auth/user-info-fetcher/portal-token.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		if (!this.idToken) {
+			return this.returnError(StatusCodes.UNAUTHORIZED, `Unauthorized error: No idToken found in the header.`);
+		}
+
+		// Check the idToken, provided in header
+		const idTokenVerification = await this.verifyIdToken(oauthProvider);
+		if (idTokenVerification.error) {
+			return idTokenVerification;
+		}
+		// return this.returnOk();
```
Comment (Eengineer1) [inline src/middleware/auth/user-info-fetcher/portal-token.ts:33]: Looks redundant + safe to remove.

File: src/services/track/helpers.ts [hunk 2]
```diff
@@ -31,3 +34,19 @@ export function isDIDTrack(data: TrackData): data is IDIDTrack {
 export function toCoin(amount: bigint, denom = MINIMAL_DENOM): Coin {
 	return coin(amount.toString(), denom);
 }
+
+export function builSubmitOperation (subscription: Stripe.Subscription, name: string, options?: ISubmitOptions) {
```
Comment (Eengineer1) [inline src/services/track/helpers.ts:38]: Marking typo to resolve:

```suggestion
export function buildSubmitOperation (subscription: Stripe.Subscription, name: string, options?: ISubmitOptions) {
```

**Review by Eengineer1 — APPROVED** [review]
Looks well polished.

Approved.

**Pull request merged**